test/utils/adjustPrecision.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { adjustPrecision } from 'utils'
+
+describe('bellow precision', () => {
+  test('with decimals', () => {
+    expect(adjustPrecision('1.1', 3)).toBe('1.1')
+  })
+  test('only integers', () => {
+    expect(adjustPrecision('1', 3)).toBe('1')
+  })
+  test('partial number', () => {
+    expect(adjustPrecision('1.', 3)).toBe('1.')
```

Comment:
This is a bit odd, isn't it?

I see JS accept `1.` as input, but would we want that?

Reply:
Because the validation runs on every keystroke. I don't want to prevent the user from inserting a decimal value by preventing him to place a dot.

Reply:
aha, okay!